cpp/oneapi/dal/backend/memory.hpp

```diff
@@ -112,23 +112,24 @@ inline T* malloc_host(const sycl::queue& queue, std::int64_t count) {
     return malloc<T>(queue, count, sycl::usm::alloc::host);
 }
 
-inline sycl::event memcpy(sycl::queue& queue, void* dest, const void* src, std::size_t size) {
-    ONEDAL_ASSERT(size > 0);
-    return queue.memcpy(dest, src, size);
-}
-
 inline sycl::event memcpy(sycl::queue& queue,
                           void* dest,
                           const void* src,
                           std::size_t size,
                           const event_vector& deps) {
     ONEDAL_ASSERT(size > 0);
+    ONEDAL_ASSERT(is_known_usm(queue, dest));
+    ONEDAL_ASSERT(is_known_usm(queue, src));
     return queue.submit([&](sycl::handler& cgh) {
         cgh.depends_on(deps);
         cgh.memcpy(dest, src, size);
     });
 }
 
+inline sycl::event memcpy(sycl::queue& queue, void* dest, const void* src, std::size_t size) {
```

Comment:
Why not to add default value to `const event_vector& deps` in the function above?